src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -510,16 +599,25 @@ public with sharing class PMT_PaymentWizard_CTRL {
     public PageReference createPayments() {
     	Savepoint sp = Database.setSavepoint();
         try {
-	        // first delete unpaid payments
-	        npe01__oppPayment__c[] old = [SELECT Id FROM npe01__oppPayment__c WHERE npe01__opportunity__c = :currentOpp.id AND npe01__paid__c = false];
-	        if (!old.isEmpty()) {
-                delete old;
+            List<npe01__oppPayment__c> paymentsToDelete = new List<npe01__OppPayment__c>(); 
+            List<npe01__OppPayment__c> existingPayments = [
+                SELECT Id, npe01__Paid__c
+                FROM npe01__OppPayment__c
+                WHERE npe01__Opportunity__c = :currentOpp.Id
+            ];
+
+            for (npe01__OppPayment__c payment : existingPayments) {
+                if (payment.npe01__Paid__c == false) {
```

Comment:
I'm not sure if this a different bug, but this will delete any payments that have been written off already. Given this scenario:

Opportunity Amount = $1000
Payment 1 for $100 - Written Off
Payment 2 for $100 - Paid

Click Schedule Payments
Create 8 payments, 1 a month
Click Calculate Payments and the Create Payments

Payment 1 that was written off is deleted.
Payment 2 that was paid remains.
8 Payments created for $100 each.
Total payments is only $900

Reply:
Good catch - I just checked the current page and it is wrong as well.  I've updated this as I agree with you - I believe it is wrong.